### PR TITLE
Release 24 April 2024

### PR DESCRIPTION
**Pull request by Sachinbisht27**
The release contains the changes for:
- #1 
- #2 
- #3 
- #6 
- #7 
- #8 
- #9

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    if: github.ref == 'refs/heads/develop'
+    steps:
+    - uses: 'actions/checkout@v4'
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v2'
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+    - name: Debug GCP credentials
```
Comment (Satendra-SR) [inline .github/workflows/build.yml:21]: This is not a debugging step @Sachinbisht27. We are copying the content from Github Environment variable to a new file `credentials.json` and using it for connecting with GCP in a few next steps.

```suggestion
    - name: Create credentials.json file using Github Environment variable
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: 'actions/checkout@v4'
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v2'
+        with:
+          credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+      - name: Debug GCP credentials
```
Comment (Satendra-SR) [inline .github/workflows/build.yml:83]: Same here

**Review by Satendra-SR — APPROVED** [review]
Few changes requires, else looks good to go

File: .github/workflows/build.yml (hunk 1 above)
Comment (Sachinbisht27) [reply .github/workflows/build.yml]: Same we are doing in the Cloud SDK step so we can remove this step also.

**Pull request merged**